esdt-safe/src/lib.rs

```diff
@@ -243,16 +270,22 @@ pub trait EsdtSafe:
 
         self.require_below_max_amount(&payment_token, &payment_amount);
 
+        // This addr is used for the refund, if the transaction fails
+        // This is passed by the BridgeTokenWrapper contract
+        let user_addr = match opt_refund_address {
+            OptionalValue::Some(addr) => addr,
```

Comment:
also self.create_transaction_event( should be updated so we can link the transaction to a refund one

Reply:
I think it is still better to flag that the transaction is a refund, as it will be way easier for the api to link the transactions